src/sema/types.rs

```diff
@@ -577,6 +577,12 @@ fn event_decl(
                 loc: name.loc,
             })
         } else {
+            if ns.target.is_substrate() && field.indexed {
+                ns.diagnostics.push(Diagnostic::error(
+                    field.loc,
+                    "indexed event fields must have a name on substrate".into(),
```

Comment:
It would be great to include a link into the documentation here which explains how topics are encoded on Substrate.

Reply:
So how about writing some documentation for Solang on this then. This would help users.

Reply:
I tend to disagree: As a user, why should I care at all how the topics are working under the hood? The only thing I'd care about is that they are compatible with `ink!` tooling. I guess that's probably the reason ink doesn't mention it in their docs as well. Furthermore, if this goes into our docs, we are providing and maintaining documentation of random parts of the  `ink!` on-chain engine, which in my opinion is something we should avoid heavily.